### ISSUE TITLE
Building the SDK fails on Windows fails trying to execute python3

### ISSUE THREAD

**Issue by pearson** [dart-lang/sdk#46045]
I'm attempting to build the dart-lang SDK under Windows 10. As far as I can tell I have the environment configured as per the documentation at [Building](https://github.com/dart-lang/sdk/wiki/Building). These builds fail trying to execute python3. (Or, at least, I assume it is the python step that causes the overall build failure.)

My command:
`tools\build.py --no-goma --mode release --arch x64 create_sdk`

The error:
```
ERROR at //sdk_args.gni:22:24: Could not execute interpreter.
  default_git_folder = exec_script("$_dart_root/tools/get_dot_git_folder.py",
                       ^----------
I was trying to execute "python3".
See //BUILD.gn:6:1: whence it was imported.
import("sdk_args.gni")
^--------------------
Command failed: C:\Users\Chris\Desktop\dart-sdk\sdk\buildtools\gn.exe gen out\ReleaseX64 --ide=vs --args=target_os="win" host_cpu="x64" target_cpu="x64" dart_target_arch="x64" dart_use_compressed_pointers=false dart_use_crashpad=false dart_snapshot_kind="app-jit" dart_use_fallback_root_certificates=true dart_use_tcmalloc=false is_debug=false is_release=true is_product=false dart_debug=false dart_runtime_mode="develop" exclude_kernel_service=false is_clang=true dart_vm_code_coverage=false is_asan=false is_lsan=false is_msan=false is_tsan=false is_ubsan=false is_qemu=false dart_platform_sdk=false use_goma=false goma_dir="None" verify_sdk_hash=true
ninja -C out\ReleaseX64 create_sdk
ninja: error: loading 'build.ninja': The system cannot find the file specified.

ninja: Entering directory `out\ReleaseX64'
BUILD FAILED
```

I believe the environment is set up correctly:

- All steps are run as admin
- Visual Studio is installed
- The Windows 10 SDK is installed, and the Debugging Tools for Windows were added to it.
- The source was retrieved using `fetch dart`.
- The Chromium depot tools were installed.
- I ensured that `python.bat` (as well as `python3.bat`) were at the front of my path. I double-checked this by adding `echo` lines to the start of the `bat` files and testing `python` and `python3` on the command line. The echo lines appeared, so the path seems correct.

Any suggestions as to what might be wrong here? Thanks!

**Comment by athomas** [comment]
Are you running this in a cmd.exe shell or PowerShell? cmd.exe is the one that should work (preferably as administrator, though it should not be necessary after fetch dart).

Some things to try:
```
where python3
where ninja
where gclient
gclient sync
python3 tools/build.py --no-goma --mode release --arch x64 create_sdk
```

**Comment by pearson** [comment]
@athomas , thank you for your reply, but I simply cannot replicate this problem. It's bizarre, as I'd fought with it for quite a while that day.

Just in case you're interested... The only potential difference I can see is that, between the failed and successful builds of the Dark SDK, I tried to build the Flutter SDK. An unrelated issue led to failed builds, and I followed the instructions [in this post](https://github.com/flutter/flutter/issues/28720#issuecomment-660631723) and successfully built the Flutter SDK. However, there doesn't seem to be any significant differences in the configuration of the tools, git, or the environment variables, and I haven't been successful in re-creating the problem by tweaking them.
